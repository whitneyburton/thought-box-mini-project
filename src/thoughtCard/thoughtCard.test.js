import React from 'react';
import { shallow } from 'enzyme';
import ThoughtCard from './ThoughtCard.js';

describe('ThoughtCard', () => {

  it('should match the snapshot', () => {
    const mockThought = {};
    const wrapper = shallow(<ThoughtCard thought={mockThought} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should start with the appropriate default state', () => {
    const wrapper = shallow(<ThoughtCard />);
    expect(wrapper.state('editable')).toEqual(false);
    expect(wrapper.state('title')).toEqual('');
    expect(wrapper.state('body')).toEqual('');
  })

  it('should delete a thought when the delete button is clicked', () => {
    const mockThought = {};
    const mockDeleteClick = jest.fn();
    const wrapper = shallow(<ThoughtCard deleteThought={mockDeleteClick} thought={mockThought} />);
    wrapper.find('.delete-button').simulate('click');
    expect(mockDeleteClick).toHaveBeenCalled();
  });

  it('should turn the editable property boolean to true when the edit button is clicked', () => {
    const wrapper = shallow(<ThoughtCard />);
    expect(wrapper.state('editable')).toEqual(false);
    wrapper.find('.edit-button').simulate('click');
    expect(wrapper.state('editable')).toEqual(true);
  });

  it('should turn the editable property boolean to false when the save button is clicked', () => {
    const wrapper = shallow(<ThoughtCard />);
    wrapper.setState({ editable: true });
    wrapper.find('.edit-button').simulate('click');
    expect(wrapper.state('editable')).toEqual(false);
  });

  it('should update the state properties title and body onChange of edit inputs', () => {
    const wrapper = shallow(<ThoughtCard />);
    const expectedState = {
      title: '',
      body: '',
      editable: false,
    };

    expect(wrapper.state()).toEqual(expectedState);
    wrapper.find('.edit-button').simulate('click');
    expect(wrapper.state('editable')).toEqual(true);
    wrapper.instance().updateInfo({target: {name: 'title', value: 'new idea'}});
    expect(wrapper.state('title')).toEqual('new idea');
  });

});