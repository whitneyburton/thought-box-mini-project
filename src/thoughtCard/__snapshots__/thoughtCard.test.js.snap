// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThoughtCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThoughtCard
    thought={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="title"
          contentEditable={false}
        />,
        <p
          className="body"
          contentEditable={false}
        />,
        <div
          className="buttons-container"
        >
          <button
            className="edit-button buttons"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="delete-button buttons"
          >
            Delete
          </button>
        </div>,
      ],
      "className": "ThoughtCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "title",
          "contentEditable": false,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "body",
          "contentEditable": false,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="edit-button buttons"
              onClick={[Function]}
            >
              Edit
            </button>,
            <button
              className="delete-button buttons"
            >
              Delete
            </button>,
          ],
          "className": "buttons-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Edit",
              "className": "edit-button buttons",
              "id": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Edit",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Delete",
              "className": "delete-button buttons",
              "id": undefined,
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Delete",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="title"
            contentEditable={false}
          />,
          <p
            className="body"
            contentEditable={false}
          />,
          <div
            className="buttons-container"
          >
            <button
              className="edit-button buttons"
              onClick={[Function]}
            >
              Edit
            </button>
            <button
              className="delete-button buttons"
            >
              Delete
            </button>
          </div>,
        ],
        "className": "ThoughtCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "title",
            "contentEditable": false,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "body",
            "contentEditable": false,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="edit-button buttons"
                onClick={[Function]}
              >
                Edit
              </button>,
              <button
                className="delete-button buttons"
              >
                Delete
              </button>,
            ],
            "className": "buttons-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edit",
                "className": "edit-button buttons",
                "id": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Edit",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Delete",
                "className": "delete-button buttons",
                "id": undefined,
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "Delete",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
